Added some more resources (all free) to the list

ayra-jafri opened the pull request:
Added a few free resources in the tutorials section (added the Helsinki course, the Java2s book, the Freeware Java tutorials, and the Cave of Programming's course. Also added the subreddit r/dailyprogrammers in the challenges section, and added r/javahelp and r/javaexamples under the community section. 

<!-- Please follow this checklist and put an x in each of the boxes, like this: [x]. It will ensure that our team takes your pull request seriously. -->

- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] My article does not contain shortened URLs or affiliate links.

If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.

Closes #XXXXX

ayra-jafri committed:
Added some more resources (all free) to the list

huyenltnguyen reviewed (changes requested):
@ayra-jafri Thank you for the PR!

Your changes look good overall. However, there are some small changes that need to be made to conform with the [Style Guide](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/docs/style-guide-for-guide-articles.md#links).

Once you have completed those suggested changes, we will be happy to merge your PR.

ayra-jafri committed:
Updated the index.md and removed a few links

Specifically removed the Java2s book link, and the Java Freeware link. Also changed the link for the Helsinki MOOC to https.

huyenltnguyen reviewed (approved):
LGTM 👍 

@ayra-jafri Thank you for the PR and for the additional changes ✨ 🎉 We're happy to accept the PR and look forward to your future contributions.

Happy coding!
